client/my-sites/post-selector/selector.jsx

```diff
@@ -444,11 +446,18 @@ const PostSelectorPosts = React.createClass( {
 
 export default connect( ( state, ownProps ) => {
 	const { siteId, query } = ownProps;
+
+	const queryWithVersion = { ...query };
+	if ( ! isJetpackSite( state, siteId ) || isJetpackMinimumVersion( state, siteId, '99' ) ) {
+		queryWithVersion.apiVersion = '1.2';
+	}
```

Comment:
Nitpicky, but since const assignment and property assignment happen at the same time, I'd condense into:

```js
const apiVersion = ( ! … || … )
  ? '1.2'
  : undefined;
const queryWithVersion = { ...query, apiVersion };
```